github/challenge_processing_script.py

```diff
@@ -0,0 +1,106 @@
+import os
+import requests
+import sys
+
+from config import *
+from utils import (
+	create_issue_in_repo,
+	comment_on_pr,
+	construct_challenge_zip_file,
+	get_request_header,
+	load_host_configs,
+	validate_token,
+)
+
+sys.dont_write_bytecode = True
+
+HOST_AUTH_TOKEN = None
+CHALLENGE_HOST_TEAM_PK = None
+EVALAI_HOST_URL = None
+
+os.environ["CHALLENGE_ERRORS"] = "False"
+
+IS_VALIDATION = os.getenv("IS_VALIDATION")
+GITHUB_EVENT_NAME = os.getenv("GITHUB_EVENT_NAME")
+
+GITHUB_CONTEXT = json.loads(os.getenv("GITHUB_CONTEXT"))
+GITHUB_AUTH_TOKEN = os.getenv("GITHUB_AUTH_TOKEN")
+
+if GITHUB_EVENT_NAME.startswith("pull_request"):
+	PR_NUMBER = GITHUB_CONTEXT["event"]["number"]
+
+if __name__ == "__main__":
+
+	if IS_VALIDATION == "False" and GITHUB_EVENT_NAME.startswith("pull_request"):
+		sys.exit(0)
+
+	configs = load_host_configs(HOST_CONFIG_FILE_PATH)
+	if configs:
+		HOST_AUTH_TOKEN = configs[0]
+		CHALLENGE_HOST_TEAM_PK = configs[1]
+		EVALAI_HOST_URL = configs[2]
+	else:
+		sys.exit(1)
+
+	# Fetching the url
+	if IS_VALIDATION == "True":
+		url = "{}{}".format(EVALAI_HOST_URL, CHALLENGE_CONFIG_VALIDATION_URL.format(CHALLENGE_HOST_TEAM_PK))
+	if IS_VALIDATION == "False" and GITHUB_EVENT_NAME == "push":
+		url = "{}{}".format(EVALAI_HOST_URL, CHALLENGE_CREATE_OR_UPDATE_URL.format(CHALLENGE_HOST_TEAM_PK))
+
+	headers = get_request_header(HOST_AUTH_TOKEN)
+
+	# Creating the challenge zip file and storing in a dict to send to EvalAI
+	construct_challenge_zip_file(CHALLENGE_ZIP_FILE_PATH, IGNORE_DIRS, IGNORE_FILES)
+	zip_file = open(CHALLENGE_ZIP_FILE_PATH, 'rb')
+	file = {"zip_configuration": zip_file}
+
+	data = {"GITHUB_REPOSITORY": GITHUB_REPOSITORY} 
+
+	try:
+		response = requests.post(url, data=data, headers=headers, files=file)
+
+		if response.status_code != http.HTTPStatus.OK and response.status_code != http.HTTPStatus.CREATED:
+			response.raise_for_status()
+		else:
+			print("\n"+response.json()["Success"])
+	except requests.exceptions.HTTPError as err:
+		if response.status_code in EVALAI_ERROR_CODES:
+			is_token_valid = validate_token(response.json())
+			if is_token_valid:
+				error = response.json()["error"]
+				error_message = "\nThere was were some errors in the challenge config:\n{}".format(error)
+				print(error_message)
+				os.environ["CHALLENGE_ERRORS"] = error_message
+		else:
+			print("\nThere was an error: {}".format(err))
+			os.environ["CHALLENGE_ERRORS"] = str(err)
+	except Exception as e:
+		if IS_VALIDATION=="True":
+			error_message = "\nThere was an error when validating the challenge config: {}".format(e)
+			print(error_message)
+			os.environ["CHALLENGE_ERRORS"] = error_message
+		else:
+			error_message = "\nThere was an error: {}".format(e)
+			print(error_message)
+			os.environ["CHALLENGE_ERRORS"] = error_message
+
+	zip_file.close()
+	os.remove(zip_file.name)
+
+	if os.environ.get("CHALLENGE_ERRORS") != "False" and IS_VALIDATION=="True" and GITHUB_EVENT_NAME.startswith("pull_request"):
```

Comment:
can we put these conditions in separate methods with appropriate naming? Currently, I am not able to understand which cases are we trying to handle. It will be clear if we have separate method for these checks